doc/services/pm/device.rst

```diff
@@ -158,8 +158,45 @@ a particular device. It is important to emphasize that, although the
 state is tracked by the subsystem, it is the responsibility of each device driver
 to handle device actions(:c:enum:`pm_device_action`) which change device state.
 
-Each :c:enum:`pm_device_action` have a direct an unambiguous relationship with
-a :c:enum:`pm_device_state`.
+Each :c:enum:`pm_device_state` state is defined as follows:
+
+- ``OFF``: Device hardware is not powered. Once this state is exited, the device
+  is considered to be in its reset state. When a device driver is initialized, the
+  device is presumed to be in this state, it is the initial state. We assume that
+  the device has been left in this state, or the ``SUSPENDED`` state, by the
+  previous "owner" of the device, be that a bootloader, an SoC (reset), or a
+  deinitialized device driver. If the device was left in the ``SUSPENDED`` state,
+  the :c:enum:`pm_device_action` ``TURN_ON`` action will be invoked whether or
+  not the device hardware lost power.
+
```

Comment:
The above all looks technically correct, but is pretty dense in terms of what a user needs to actually do.
Maybe add something like the following?
```
As a result, `PM_DEVICE_ACTION_TURN_ON` should be able to transition the device from any internal state into `SUSPENDED`, since no guarantees can be made across resets. This is typically achieved through toggling a reset pin or triggering a software reset through a register write before performing any additional configuration needed to meet the requirements of ``SUSPENDED``.
```

Reply:
Added this unaltered aside from .rst formatting, removed preceeding duplicate information, appended

```
For devices where this is not possible, the device driver
must presume the device is in either the ``OFF`` or ``SUSPENDED`` state at
time of initialization, as these are the states within which device drivers
may be deinitialized.
```
since this is the case for some nordic hardware, we don't have a way of resetting UARTs outside of system resets, so bootloaders need to do the deinit properly, and app needs to presume bootloader nicely reset the hardware :)